core/src/main/java/org/bitcoinj/core/Block.java

```diff
@@ -930,7 +930,7 @@ public Date getTime() {
     @VisibleForTesting
     public void setTime(Instant time) {
         unCacheHeader();
-        this.time = time.truncatedTo(ChronoUnit.SECONDS); // convert to Bitcoin time
+        this.time = TimeUtils.checkBitcoinTime(time);
```

Comment:
throw or truncate?

Reply:
Out of scope, but related: we should add `time` to the constructors and deprecate this method.

Reply:
Absolute! I believe that's part of the Message hierarchy rework.